# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/cocoa/nsWidgetFactory.mm
# Commit: 8287c3d0c106
# Full Hash: 8287c3d0c1064096ee7c162060e6615d72722ee0
# Author: Christoph Walcher <christoph-wa@gmx.de>
# Date: 2019-06-20 09:46:31
# Regressor Bug: 909760
# File Overlap Count: 1
# Description:
#   Bug 909760 - Show download progress in the MacOS Finder r=spohl,mak
#   
#   Show download progress in the MacOS Finder
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D30688
# ==============================================================================

diff -r 5b9a3de04646 -r 8287c3d0c106 widget/cocoa/nsWidgetFactory.mm
--- a/widget/cocoa/nsWidgetFactory.mm	Wed Jun 19 18:39:02 2019 +0000
+++ b/widget/cocoa/nsWidgetFactory.mm	Wed Jun 19 18:48:49 2019 +0000
@@ -82,6 +82,9 @@
 #include "nsMacDockSupport.h"
 NS_GENERIC_FACTORY_CONSTRUCTOR(nsMacDockSupport)
 
+#include "nsMacFinderProgress.h"
+NS_GENERIC_FACTORY_CONSTRUCTOR(nsMacFinderProgress)
+
 #include "nsMacSharingService.h"
 NS_GENERIC_FACTORY_CONSTRUCTOR(nsMacSharingService)
 
@@ -123,6 +126,7 @@
 NS_DEFINE_NAMED_CID(NS_SYSTEMALERTSSERVICE_CID);
 NS_DEFINE_NAMED_CID(NS_NATIVEMENUSERVICE_CID);
 NS_DEFINE_NAMED_CID(NS_MACDOCKSUPPORT_CID);
+NS_DEFINE_NAMED_CID(NS_MACFINDERPROGRESS_CID);
 NS_DEFINE_NAMED_CID(NS_MACSHARINGSERVICE_CID);
 NS_DEFINE_NAMED_CID(NS_MACWEBAPPUTILS_CID);
 NS_DEFINE_NAMED_CID(NS_STANDALONENATIVEMENU_CID);
@@ -153,6 +157,7 @@
     {&kNS_SYSTEMALERTSSERVICE_CID, false, NULL, OSXNotificationCenterConstructor},
     {&kNS_NATIVEMENUSERVICE_CID, false, NULL, nsNativeMenuServiceXConstructor},
     {&kNS_MACDOCKSUPPORT_CID, false, NULL, nsMacDockSupportConstructor},
+    {&kNS_MACFINDERPROGRESS_CID, false, NULL, nsMacFinderProgressConstructor},
     {&kNS_MACSHARINGSERVICE_CID, false, NULL, nsMacSharingServiceConstructor},
     {&kNS_MACWEBAPPUTILS_CID, false, NULL, nsMacWebAppUtilsConstructor},
     {&kNS_STANDALONENATIVEMENU_CID, false, NULL, nsStandaloneNativeMenuConstructor},
@@ -182,6 +187,7 @@
     {"@mozilla.org/system-alerts-service;1", &kNS_SYSTEMALERTSSERVICE_CID},
     {"@mozilla.org/widget/nativemenuservice;1", &kNS_NATIVEMENUSERVICE_CID},
     {"@mozilla.org/widget/macdocksupport;1", &kNS_MACDOCKSUPPORT_CID},
+    {"@mozilla.org/widget/macfinderprogress;1", &kNS_MACFINDERPROGRESS_CID},
     {"@mozilla.org/widget/macsharingservice;1", &kNS_MACSHARINGSERVICE_CID},
     {"@mozilla.org/widget/mac-web-app-utils;1", &kNS_MACWEBAPPUTILS_CID},
     {"@mozilla.org/widget/standalonenativemenu;1", &kNS_STANDALONENATIVEMENU_CID},